akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -455,13 +456,29 @@ final case class `Content-Location`(uri: Uri) extends jm.headers.ContentLocation
  * two there is slight but important difference regarding how parameter values are formatted. In RFC6266 parameters values are without quotes and
  * in RFC2616 they are quoted. Since common practice among http servers is to understand quoted values, we use older document
  * as reference here.
+ * Extended (i.e. encoded) parameter values are not quoted, see ext-parameter - https://tools.ietf.org/html/rfc5987#section-3.2.1
  */
 object `Content-Disposition` extends ModeledCompanion[`Content-Disposition`]
 final case class `Content-Disposition`(dispositionType: ContentDispositionType, params: Map[String, String] = Map.empty)
   extends jm.headers.ContentDisposition with RequestResponseHeader {
   def renderValue[R <: Rendering](r: R): r.type = {
+    import Rendering.{ contentDispositionFilenameSafeChars => safeChars }
     r ~~ dispositionType
-    params foreach { case (k, v) => r ~~ "; " ~~ k ~~ '=' ~~#! v }
+    val renderExtFilename =
+      params.get("filename").map(safeChars.indexOfFirstMismatch(_) >= 0) getOrElse false
+    val withExtParams =
+      if (renderExtFilename && !params.contains("filename*"))
+        params + ("filename*" -> params("filename"))
+      else params
+    withExtParams foreach {
+      case (k, v) if k == "filename" =>
+        r ~~ "; " ~~ k ~~ '=' ~~ '"'
+        r.putReplaced(v, keep = safeChars, placeholder = '?') ~~ '"'
+      case (k, v) if k endsWith "*" =>
```

Comment:
I guess this whole logic should be moved to a more general place for wider adoption, but for this PR we can leave it here.